File: app/src/main/java/com/mithril/mobilegoldenleaf/database/converter/CaledarConverter.java
package com.mithril.mobilegoldenleaf.database.converter;

import androidx.room.TypeConverter;

import java.util.Calendar;

public class CaledarConverter {

    @TypeConverter
    public Long toLong(Calendar calendar) {
        return calendar.getTimeInMillis();
    }

    @TypeConverter
    public Calendar toCalendar(Long value) {
        Calendar currentyMonment = Calendar.getInstance();
        if (value != null) {
            currentyMonment.setTimeInMillis(value);
        }
        return currentyMonment;
    }

}

File: app/src/main/java/com/mithril/mobilegoldenleaf/persistence/converter/CaledarConverter.kt
package com.mithril.mobilegoldenleaf.persistence.converter

import androidx.room.TypeConverter

import java.util.Calendar

class CaledarConverter {

    @TypeConverter
    fun toLong(calendar: Calendar?): Long? {
        return calendar?.timeInMillis
    }

    @TypeConverter
    fun toCalendar(value: Long?): Calendar {
        val currentyMonment = Calendar.getInstance()
        if (value != null) {
            currentyMonment.timeInMillis = value
        }
        return currentyMonment
    }

}
